### PR TITLE
Add CopyUpToDate ETW

### PR DESCRIPTION
**Pull request by Forgind**
Progress towards #6658 

Adds an up-to-date ETW for the Copy task. Also captures metadata copying just because that happens if it was up-to-date. (Should we have a check rather than always copying?)

### CODE REVIEW

**Review by rainersigwald — COMMENTED** [review]
Is this the right place for these events, to catch the "incremental build copying the same outputs to the same locations" case? I naïvely assumed it would be in `DoCopyIfNecessary`.

**Comment by Forgind** [comment]
The part I put ETW traces around looks like "the file at destination was copied from source." DoCopyIfNecessary has "files are the same size and timestamp" and "files have the same name." There's a fourth check buried in DoCopyWithRetries that does a deep file comparison.

Looking at those four, the first sounded sufficient for if it's just an incremental build, and I stopped there because there deeper I try to push it, the more complicated it gets. This is a simple check with no code movement. To add the next level, I'd need three possible ending places, one of which is currently also executed when actually copying, so I would need a little extra control logic there. The last check is in a catch. I think it would be impossible to hit without capturing copies as well without substantial code changes.

I can check to see if my assumption about the first being sufficient is true or just put in the change to capture the first two levels. I don't think it's worth it to try for the third.

**Comment by rainersigwald** [comment]
> I can check to see if my assumption about the first being sufficient is true

Sounds like a plan.

File: src/Tasks/Copy.cs [hunk 1]
```diff
@@ -561,6 +568,10 @@ int parallelism
                                 success = false;
                             }
                         }
+                        else
+                        {
+                            MSBuildEventSource.Log.CopyUpToDateStop(destItem.ItemSpec);
```
Comment (Forgind) [inline src/Tasks/Copy.cs:573]: Specify whether it was up to date or not

**Review by rainersigwald — APPROVED** [review]

**Review by ladipro — APPROVED** [review]

**Review by AR-May — APPROVED** [review]

**Pull request merged**